File: src/main/java/org/shark/example/controller/emqx/EmqxController.java
package org.shark.example.controller.emqx;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.shark.example.service.base.pojo.ResponseDto;
import org.shark.example.service.emqx.SendMessageService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController
@RequestMapping("/emqx")
@Slf4j
public class EmqxController {

    private final SendMessageService gsendMessageServicetI18nService;

    @GetMapping
    public ResponseDto<Void> getMessage(@RequestParam String message) {
        return gsendMessageServicetI18nService.start(message);
    }
}

File: src/main/kotlin/org/shark/example/controller/emqx/EmqxController.kt
package org.shark.example.controller.emqx

import org.shark.example.service.base.pojo.ResponseDto
import org.shark.example.service.emqx.SendMessageService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import org.slf4j.LoggerFactory

@RestController
@RequestMapping("/emqx")
class EmqxController(
    private val sendMessageService: SendMessageService
) {
    private val log = LoggerFactory.getLogger(EmqxController::class.java)

    @GetMapping
    fun getMessage(@RequestParam message: String): ResponseDto<Void> {
        return sendMessageService.start(message)
    }
}
